src/web/wms/src/main/java/org/geoserver/wms/web/publish/WMSLayerConfig.java

```diff
@@ -203,12 +204,20 @@ private void initWMSCascadedUI(IModel<LayerInfo> layerModel) {
 
         WMSLayerInfo wmsLayerInfo = (WMSLayerInfo) layerModel.getObject().getResource();
         // for new only
-        if (layerModel.getObject().getId() == null) wmsLayerInfo.reset();
-        else {
-            wmsLayerInfo.getAllAvailableRemoteStyles().clear();
+        try {
+            if (layerModel.getObject().getId() == null) wmsLayerInfo.reset();
+            else {
+                // pull latest styles from remote WMS
+                wmsLayerInfo.getAllAvailableRemoteStyles().clear();
```

Comment:
This is a strange behavior for an API, your API should not require the invoker to know the internals of your class, IMHO the following methods should exist:
- ``getAllAvailableRemoteStyles()`` should return the list of all the available remote styles on the remote server, I would rename it ``getRemoteStyles``;
- ``setRemoteStyles(List<...> remoteStyles)`` sets the remote styles equal to the provided list, if the provided list is ```NULL`` it should be converted to an empty list; 
- ``addRemoteStyles(List<...> remoteStyles)`` will add the provided remote styles to the already existing remote styles;

If the remote styles are represented by a string then a ``Set`` should be used, this would avoid duplications.

Reply:
understood, will change accordingly

Reply:
understood, IMHO I see no use of `addRemoteStyles` , since we will need to match local list of styles with what is available on remote, it would be safe to replace the styles completely.

Reply:
@nmco all *Info objects in GeoServer use a "live list" approach, where you get the list and manipulate it at will. There are no other methods to manipulate lists. It's a pattern that is applied only there, but at least there, it should be consistent.